### PR TITLE
[#3964] Bandaid to fix old forms with simple conditionals

### LINKED ISSUE #3964: FE logic: When the value of number field is used to trigger visibility, components are emptied when "Clear on hide" is checked
### Product versie / Product version

2.4.3 - 2.6.0

### Omschrijf het probleem / Describe the bug

When the value of number/currency field is used in FE logic and the component that is made visible has: "Clear on hide" checked, the values entered into this field will be removed at validation (check_logic)

`if Getal = 0 show Tekstvlak`

pooof
Kooha-2024-03-01-21-23-53

If a different logic is used on the same Number component like: `if not Getal hide Tekstvlak`
This bug does not occur, so it seems to be specific to the value in the number field

# Discussion:

We changed the formio-builder so that for new forms the conditional value is not saved as a string but as a number.

However, this will not fix existing forms that have a configuration where the conditional value is already a string. For fixing these forms there were 2 possibilities:
- Migration: find forms that have this problem and fix the configuration. This migration is quite complex and would also have to run on import.
- Bandaid fix in the logic (like for checkboxes: https://github.com/open-formulieren/open-forms/blob/1d22bdc733f5fa854f4d07cfe357e62e35a82db0/src/openforms/formio/utils.py#L358). This is quite easy, although care needs to be taken with the int/float casting.

Considerations:
- Writing the migration feels like the 'right thing':  new components have fixed behaviour and you don't want to carry along a legacy fix forever. It's better to root it out as soon as possible.
- Backporting the migration is tricky

Conclusion:
- Migration for master
- Bandaid for backport

In addition:
- See if you can fix checkboxes so we can remove the bandaid fix in master

### PR DESCRIPTION
**Pull request by SilviaAmAm**
Fixes #3964 (partially)

### CODE REVIEW

File: src/openforms/submissions/tests/form_logic/test_modify_components.py [hunk 1]
```diff
@@ -1046,6 +1046,57 @@ def test_checkbox_frontend_logic(self):
...
+                        "key": "textField1",
+                        "type": "textfield",
+                        "clearOnHide": True,
+                        "conditional": {"eq": "1", "show": True, "when": "number"},
+                    },
+                    {
+                        "key": "textField2",
+                        "type": "textfield",
+                        "clearOnHide": True,
+                        "conditional": {"eq": "1", "show": True, "when": "currency"},
```
Comment (Viicos) [inline src/openforms/submissions/tests/form_logic/test_modify_components.py:1074]:
```suggestion
                        "conditional": {"eq": "1.0", "show": True, "when": "currency"},
```
Maybe a test with a `float`?

**Review by Viicos — APPROVED** [review]

File: src/openforms/submissions/tests/form_logic/test_modify_components.py (hunk 1 above)
Comment (SilviaAmAm) [reply src/openforms/submissions/tests/form_logic/test_modify_components.py:1074]: I was worried of potential flakiness due to comparing floats :thinking:

Comment (Viicos) [reply src/openforms/submissions/tests/form_logic/test_modify_components.py:1074]: I think it should be alright, as long as you don't perform any calculations on the floating numbers

And if we have flakyness in the test we will have it at runtime as well so I think this is a good thing :)

Comment (SilviaAmAm) [reply src/openforms/submissions/tests/form_logic/test_modify_components.py:1074]: Yes! Although I remeber at some point we had discouraged clients to do comparisons on with the floating points, so hopefully it shouldn't happen too often.

Comment (Viicos) [reply src/openforms/submissions/tests/form_logic/test_modify_components.py:1078]: If this becomes an issue we could probably use decimals to do the comparison

**Comment by SilviaAmAm** [comment]
Backports:
- 2.5.x: https://github.com/open-formulieren/open-forms/pull/4195
- 2.4.x: https://github.com/open-formulieren/open-forms/pull/4196

**Pull request merged**